src/Microsoft.AspNet.Http.Extensions/UriHelper.cs

```diff
@@ -48,7 +51,15 @@ public static class UriHelper
             FragmentString fragment = new FragmentString())
         {
             string combinePath = (pathBase.HasValue || path.HasValue) ? (pathBase + path).ToString() : "/";
-            return $"{scheme}://{host.ToString()}{combinePath}{query.ToString()}{fragment.ToString()}";
+
+            return new StringBuilder()
```

Comment:
Yes otherwise is allocs at size 16; then then starts doing interesting things when it overflows